pycontrails/core/vector.py

```diff
@@ -651,13 +655,26 @@ def sum(
         """
         if not vectors:
             return cls()
-        keys = vectors[0].data.keys()
-        for v in vectors[1:]:
-            if v.data.keys() != keys:
-                raise KeyError("Summands have incompatible keys.")
+
+        keys: Iterable[str]
+        if fill_value is None:
+            keys = vectors[0].data.keys()
+            for v in vectors[1:]:
+                if v.data.keys() != keys:
+                    diff = set(v.data.keys()).symmetric_difference(keys)
+                    raise KeyError(f"Summands have incompatible keys. Difference: {diff}")
+
+        else:
+            keys = set().union(*[v.data.keys() for v in vectors])
+
+        def _get(k: str, v: VectorDataset) -> np.ndarray:
```

Comment:
do we not have a VectorDataset.get() that would do this?

Reply:
You're absolutely right (I forget about that here). One subtle but important difference is that when you call `VectorDataset.get("some_key", some_default)`, the `some_default` value is eagerly computed. In other words, when you call `VectorDataset.get("some_key", some_expensive_calculation())`, the expensive calculation takes place regardless of whether "some_key" is in the vector. In this case, we end up constructing a numpy array only to throw it away.

I think I'll keep the implementation as it stands, but include a comment.

Reply:
Should we update the other one to not be eager?

Reply:
I don't think so -- the implementation gets too tricky. The lazy version would only be possible if the default argument was something like a callback that only gets evaluated if the key is not in the dict.

Something like this may work, but I'm still inclined not to include this in pycontrails. It's different from dict.get, and it may be over-engineered.

```python

class VectorDataset:
    ....

    def lazy_get(key, default):
        out = self.data.get(key, default)
        if out is default and callable(default):
            return default()
        return out
```